[2024-06-14T16:31:16.731+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-14T16:31:16.905+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.generate_summary scheduled__2024-06-10T00:00:00+00:00 [queued]>
[2024-06-14T16:31:16.943+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.generate_summary scheduled__2024-06-10T00:00:00+00:00 [queued]>
[2024-06-14T16:31:16.944+0000] {taskinstance.py:2306} INFO - Starting attempt 4 of 6
[2024-06-14T16:31:16.997+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): generate_summary> on 2024-06-10 00:00:00+00:00
[2024-06-14T16:31:17.016+0000] {standard_task_runner.py:63} INFO - Started process 1533 to run task
[2024-06-14T16:31:17.046+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Air_pollution_data_pipeline', 'generate_summary', 'scheduled__2024-06-10T00:00:00+00:00', '--job-id', '297', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmpmbnfx_px']
[2024-06-14T16:31:17.052+0000] {standard_task_runner.py:91} INFO - Job 297: Subtask generate_summary
[2024-06-14T16:31:17.211+0000] {task_command.py:426} INFO - Running <TaskInstance: Air_pollution_data_pipeline.generate_summary scheduled__2024-06-10T00:00:00+00:00 [running]> on host 6078ceb33835
[2024-06-14T16:31:17.625+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='khalid' AIRFLOW_CTX_DAG_ID='Air_pollution_data_pipeline' AIRFLOW_CTX_TASK_ID='generate_summary' AIRFLOW_CTX_EXECUTION_DATE='2024-06-10T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-06-10T00:00:00+00:00'
[2024-06-14T16:31:17.635+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-14T16:31:17.639+0000] {summary_analysis.py:13} INFO - starting summary statistics of pollutants emission
[2024-06-14T16:31:17.659+0000] {logging_mixin.py:188} WARNING - /opt/***/dags/statistical_analysis/summary_analysis.py:16 FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
[2024-06-14T16:31:17.682+0000] {summary_analysis.py:33} ERROR - error calculating pollutants summary statistics
Traceback (most recent call last):
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 18, in generate_summary
    sum_stat = input_clean_data.desc()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/generic.py", line 6299, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'desc'
[2024-06-14T16:31:17.689+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-14T16:31:17.691+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 18, in generate_summary
    sum_stat = input_clean_data.desc()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/generic.py", line 6299, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'desc'
[2024-06-14T16:31:17.715+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=Air_pollution_data_pipeline, task_id=generate_summary, run_id=scheduled__2024-06-10T00:00:00+00:00, execution_date=20240610T000000, start_date=20240614T163116, end_date=20240614T163117
[2024-06-14T16:31:17.754+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 297 for task generate_summary ('DataFrame' object has no attribute 'desc'; 1533)
[2024-06-14T16:31:17.815+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-14T16:31:17.890+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-14T16:31:17.894+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
